tests/testthat/test-expect_s3_class_linter.R

```diff
@@ -0,0 +1,97 @@
+test_that("expect_s3_class_linter skips allowed usages", {
+  # expect_s3_class doesn't have an inverted version
+  expect_lint("expect_true(!inherits(x, 'class'))", NULL, expect_s3_class_linter())
+  # NB: also applies to tinytest, but it's sufficient to test testthat
+  expect_lint("testthat::expect_s3_class(!inherits(x, 'class'))", NULL, expect_s3_class_linter())
+
+  # other is.<x> calls are not suitable for expect_s3_class in particular
+  expect_lint("expect_true(is.na(x))", NULL, expect_s3_class_linter())
+
+  # case where expect_s3_class() *could* be used but we don't enforce
+  expect_lint("expect_true(is.data.table(x))", NULL, expect_s3_class_linter())
+})
+
+test_that("expect_s3_class_linter blocks simple disallowed usages", {
+  expect_lint(
+    "expect_equal(class(x), 'data.frame')",
```

Comment:
A two-element expectation test would be nice to have.

Reply:
as in testing that `expect_equal(class(x), c("data.table", "data.frame"))` throws a lint as expected?

Reply:
Added such a test